invenio_administration/assets/semantic-ui/js/invenio_administration/api/actions.js

```diff
@@ -0,0 +1,34 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { APIRoutes } from "./routes";
+import { http } from "./config";
+
+const getResource = async (
+  apiEndpoint = undefined,
+  pid= undefined,
+) => {
+  return await http.get(APIRoutes.get(apiEndpoint, pid));
```

Comment:
Why is this function awaiting the result if it does not need it? Is it to have better debugging? Otherwise, `return await` is not necessary and has a tiny performance overhead.

see https://eslint.org/docs/latest/rules/no-return-await

Reply:
why it is not necessary ? it is an async call

Reply:
it is unnecessary, as this function is not consuming the promise's return value or its error. It's just returning the promise back.  See the `eslint` rule explanation or this [explanation](https://maximorlov.com/linting-rules-for-asynchronous-code-in-javascript/#6-no-return-await)

In this case, note that you already `await` for the result of this promise when you actually need to: 

``` javascript
async function deleteEndpoint() {
    return await http.delete()  // this returns a promise nonetheless
}

async function myFunc(){
    try {
        const result = await deleteEndpoint(); // promise's return value is consumed here
        console.log(result);
    } catch (error) {
        console.log(error);  // promise's rejection is handled here
    }
}
```

if you don't stop the event loop here, you will get an `undefined` as it will move on and won't stop to actually resolve the promise returned by `deleteEndpoint`.

Also note that not all async calls needs to be awaited, sometimes the event loop does not need to wait for the result of a promise to move on. You can have some performance gains from this concurrency. But that's another, unrelated discussion. See this [example](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function#async_functions_and_execution_order) if you are curious about promises and concurrency.